fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/agg/Count.java

```diff
@@ -82,7 +82,7 @@ public void checkLegalityBeforeTypeCoercion() {
     public void checkLegalityAfterRewrite() {
         // after rewrite, count(distinct bitmap_column) should be rewritten to bitmap_union_count(bitmap_column)
         for (Expression argument : getArguments()) {
-            if (argument.getDataType().isOnlyMetricType()) {
+            if (argument.getDataType().isObjectType()) {
                 throw new AnalysisException(Type.OnlyMetricTypeErrorMsg);
```

Comment:
the messge content (OnlyMetricTypeErrorMsg）should be updated
"Doris hll, bitmap, array, map, struct, jsonb, variant column must use ..."
array, map, struct, jsonb are not object type